pkg/gadgets/top/block-io/tracer/tracer.go

```diff
@@ -66,22 +68,30 @@ func NewTracer(config *Config, enricher gadgets.DataEnricherByMntNs,
 		done:          make(chan bool),
 	}
 
-	if err := t.start(); err != nil {
-		t.Stop()
+	if err := t.install(); err != nil {
+		t.cleanup()
 		return nil, err
 	}
 
 	statCols, err := columns.NewColumns[types.Stats]()
 	if err != nil {
-		t.Stop()
+		t.cleanup()
 		return nil, err
 	}
 	t.colMap = statCols.GetColumnMap()
 
+	go t.run(context.TODO())
```

Comment:
No, this PR doesn't change the current behaviour. Let me try to explain the two scenarios quickly:

### Using `kubectl-gadget`
#### Currently
It uses `NewTracer` to create **and start** the tracer, as `start()` calls [`run()`](https://github.com/inspektor-gadget/inspektor-gadget/blob/50835b9f6f0561e70f7226eef1ce47a676fa4327/pkg/gadgets/top/block-io/tracer/tracer.go#L198). Then, everything was stopped and cleaned up by calling `Stop()`. The `Start()` was not involved in this case. 

_NOTE: Yeah, I agree the name NewTracer wasn't accurate. I had proposed renaming it https://github.com/inspektor-gadget/inspektor-gadget/issues/905)_

#### With this PR
It will still use `NewTracer` to create and start the tracer and stop for cleanup. Take into account this path will not be used anymore after https://github.com/inspektor-gadget/inspektor-gadget/pull/1396, but it will use what I described here just for `ig`.

### Using `ig`
#### Currently
It uses `NewInstance()` to **only** create the tracer and then `Start()` and `Stop()`.

#### With this PR
It will still use `NewInstance()` to create the tracer, but `Run` will fully replace `Start()` and `Stop()`.

I hope I explained myself well 🙂. Please let me know if that's not the case.